frontend/utility/MissingFilesModel.cpp

```diff
@@ -162,11 +162,17 @@ Qt::ItemFlags MissingFilesModel::flags(const QModelIndex &index) const
 
 void MissingFilesModel::fileCheckLoop(QList<MissingFileEntry> files, QString path, bool skipPrompt)
 {
-	loop = false;
+	fileCheckLoop(files, path, skipPrompt, 0);
+}
+
+void MissingFilesModel::fileCheckLoop(QList<MissingFileEntry> files, QString path, bool skipPrompt, int depth)
+{
+	fileLoopCounter += 1;
 	QUrl url = QUrl().fromLocalFile(path);
 	QString dir = url.toDisplayString(QUrl::RemoveScheme | QUrl::RemoveFilename | QUrl::PreferLocalFile);
 
 	bool prompted = skipPrompt;
+	int depthWithoutFile = depth + 1;
```

Comment:
Nit: `depthWithoutFileMatch` is probably more expressive (because if I'm not mistaken you want to check how deep you nested your search without finding _anything_ so far).